src/core/SideNavigation/SidenavButton.tsx

```diff
@@ -12,9 +12,14 @@ import '@itwin/itwinui-css/css/side-navigation.css';
 
 export type SidenavButtonProps = {
   /**
-   * Whether the sidenav button is active.
+   * Whether the sidenav button is active,
+   * i.e. the current page corresponds to this button.
    */
   isActive?: boolean;
+  /**
+   * Whether the sidenav button only has submenu open.
+   */
+  isSubmenuOpen?: boolean;
```

Comment:
yea, and `isActive` takes preference

Reply:
couldnt we have some kind of a... single prop to choose from the values? or it would be too specific...

Reply:
`isActive` already exists as a boolean, so it would need to be backwards compatible. Could be something like `boolean | 'submenu-open'` but that looks weird. Open to suggestions

Reply:
If you think about it, it's kinda like the `indeterminate` state of a checkbox, where you can set both `element.checked` and `element.indeterminate` to be true, but `indeterminate` takes preference visually.

Reply:
I kind of liked your suggestion `boolean | 'submenu-open'` but `isActive` might not be the best prop for this. With v2 we could have `state` or something similar that accepts literal types. For now I don't know anything better, so we might go with `isSubmenuOpen`.